pkg/controller/controller.go

```diff
@@ -188,6 +191,19 @@ func (c *controller) Run(workers int, stopCh <-chan struct{}) {
 	c.instanceQueue.ShutDown()
 	c.bindingQueue.ShutDown()
 	c.pollingQueue.ShutDown()
+
+	waitGroup.Wait()
+}
+
+// createWorker creates and runs a worker thread that just processes items in the
+// specified queue. The worker will run until stopCh is closed. The worker will be
+// added to the wait group when started and marked done when finished.
+func createWorker(queue workqueue.RateLimitingInterface, resourceType string, maxRetries int, forgetAfterSuccess bool, reconciler func(key string) error, stopCh <-chan struct{}, waitGroup *sync.WaitGroup) {
```

Comment:
My thought on the goroutine above vs internally is that this is `createAndStartWorker`. Not obvious in the use context. The `go` at the calling site says "we're spawning threads in a loop" in an idiomatic way that is immediately apparent. 

The wait group sort of alerts you to it. 

I also want to say we're getting into having way to many arguments to a function, but it's probably okay for now.